### PR TITLE
[BUGFIX] Loading basis set data for larger elements

### PR DESCRIPTION
**Pull request by obliviateandsurrender**
**Context:**  Downloading basis set data for elements that had `SPD`-type orbitals raised an error, this PR fixes that. 

**Description of the Change:** Accounts for the `SPD`-type orbitals in the `orbital_map` for the `load_basisset` function.

**Benefits:** Molecules with such elements can now be built.

**Possible Drawbacks:** N/A

**Related GitHub Issues:** N/A

### CODE REVIEW

File: doc/releases/changelog-0.38.0.md [hunk 1]
```diff
@@ -436,6 +436,9 @@
 * `qml.qsvt` now works with "Wx" convention and any number of angles.
   [(#6105)](https://github.com/PennyLaneAI/pennylane/pull/6105)
 
+* Basis set data from the Basis Set Exchange library can now be loaded for elements with `SPD`-type orbitals.
+  [(#)](https://github.com/PennyLaneAI/pennylane/pull/)
```
Comment (obliviateandsurrender) [inline doc/releases/changelog-0.38.0.md:440]:
```suggestion
  [(#6159)](https://github.com/PennyLaneAI/pennylane/pull/6159)
```

**Review by soranjh — APPROVED** [review]
Thanks @obliviateandsurrender, looks good.

**Review by austingmhuang — APPROVED** [review]
Looks good to me! Just curious how come we didn't catch this bug earlier? Incomplete test cases?

**Comment by obliviateandsurrender** [comment]
@austingmhuang we recently started supporting elements beyond the second period which was tested. I tested it for larger basis sets for those elements (which is more expected), so this particular edge case with `STO-3G` kind of slipped through.

**Pull request merged**